modules/apps/frontend-taglib/frontend-taglib/src/main/resources/META-INF/resources/resource_selector/ResourceSelector.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayAlert from '@clayui/alert';
+import ClayButton from '@clayui/button';
+import ClayForm, {ClayInput} from '@clayui/form';
+import {openSelectionModal} from 'frontend-js-web';
+import React, {useState} from 'react';
+
+export default function ResourceSelector({
+	inputLabel,
+	inputName,
+	modalTitle,
+	portletNamespace,
+	resourceName: initialResourceName,
+	resourceValue: initialResourceValue,
+	selectEventName,
+	selectResourceURL,
+	showRemoveButton,
+	warningMessage,
+}) {
+	const [resourceName, setResourceName] = useState(initialResourceName);
+	const [resourceValue, setResourceValue] = useState(initialResourceValue);
+	const [showWarning, setShowWarning] = useState(!!warningMessage);
+
+	const onResourceSelect = () =>
+		openSelectionModal({
+			iframeBodyCssClass: '',
+			onSelect: (selectedItem) => {
+				if (selectedItem) {
+					setResourceValue(selectedItem.resourceId);
+					setResourceName(selectedItem.resourceName);
+					setShowWarning(false);
+				}
+			},
+			selectEventName: `${portletNamespace}${selectEventName}`,
+			title: modalTitle,
+			url: selectResourceURL,
+		});
+
+	return (
+		<ClayForm.Group>
+			<ClayForm.Group>
```

Comment:
It looks a bit weird to have a form group inside a form group. Can't we use some other layout component on the top? Maybe a form? (it looks like a form)

Reply:
This is not needed.

Reply:
Changed with a form

Reply:
I've chenged this with a div, because if I change it with a form, the input hidden inside the form does not send the value to the back